File: src/main/java/org/homework20241215/shop/Shop.java
package org.homework20241215.shop;


import java.util.Map;

public class Shop {

    public static ShopBuilder builder() {
        return new ShopBuilder();
    }

    private final String name;
    private Map<Item, Integer> items;

    public Shop(String name, Map<Item, Integer> items) {
        this.name = name;
        this.items = items;
    }

    public void printItems() {
        System.out.println("Shop: " + name);
        items.forEach((item, count) -> System.out.println(item + " - " + count));
    }
}

File: src/main/java/org/homework20241222/shop/Shop.kt
package org.homework20241222.shop

// Shopの()の中に記載されている変数で変数宣言されている？
//　コンストラクタ作成しなくていい？
class Shop(private val name: String, private val items: Map<Item, Int>) {

    fun printItems() {
        println(name)
        items.forEach { (item: Item, count: Int) -> println(item.toString() + ": " + count + "個") }
    }

    companion object {
        fun builder(): ShopBuilder {
            return ShopBuilder()
        }
    }

}
